common/src/main/java/com/alibaba/nacos/common/remote/client/grpc/GrpcClient.java

```diff
@@ -325,20 +376,83 @@ public Connection connectToServer(ServerInfo serverInfo) {
                 ConnectionSetupRequest conSetupRequest = new ConnectionSetupRequest();
                 conSetupRequest.setClientVersion(VersionUtils.getFullClientVersion());
                 conSetupRequest.setLabels(super.getLabels());
-                conSetupRequest.setAbilities(super.clientAbilities);
+                // set ability table
+                conSetupRequest.setAbilityTable(NacosAbilityManagerHolder.getInstance().getCurrentNodeAbilities());
                 conSetupRequest.setTenant(super.getTenant());
                 grpcConn.sendRequest(conSetupRequest);
+                // wait for response
+                RecAbilityContext synResponse = markForSetup.get(connectionId);
+                if (synResponse != null) {
+                    // try to wait for notify response
+                    synResponse.blocker.await(5000L, TimeUnit.MICROSECONDS);
+                }
+                // leave for adapting old version server
                 //wait to register connection setup
                 Thread.sleep(100L);
```

Comment:
there is no need to sleep 100ms in "waiting  setup ack"  mode

Reply:
fix